controllers/etcd_controller.go

```diff
@@ -587,6 +644,17 @@ func (r *EtcdReconciler) reconcileStatefulSet(cm *corev1.ConfigMap, svc *corev1.
 			return nil, err
 		}
 
+		// delete PVC on etcd scale down based on volumeRetentionPolicy
+		if handleScaleDown {
+			pvcList := &corev1.PersistentVolumeClaimList{}
+			if pvcList, err = r.prepareEtcdForScaleDown(etcd, ss, svc); err != nil {
+				return nil, err
+			}
+			if etcd.Spec.VolumeRetentionPolicy == nil || *etcd.Spec.VolumeRetentionPolicy == druidv1alpha1.VolumeRetentionPolicyDelete {
+				defer r.removePVCs(pvcList.Items)
```

Comment:
What if there are errors while deleting PVCs? Shouldn't we `RequeueAfter`?

Reply:
Nice catch. I'll enhance the error handling here